lib/Dialect/Moore/MooreTypes.cpp

```diff
@@ -197,143 +127,31 @@ namespace moore {
 namespace detail {
 struct IntTypeStorage : TypeStorage {
   using KeyTy = unsigned;
-  using Kind = IntType::Kind;
 
-  IntTypeStorage(KeyTy key) : kind(static_cast<Kind>(key)) {}
-  static KeyTy pack(Kind kind) { return static_cast<unsigned>(kind); }
-  bool operator==(const KeyTy &key) const { return pack(kind) == key; }
+  IntTypeStorage(KeyTy key)
+      : width(key >> 1), domain(static_cast<Domain>((key >> 0) & 1)) {}
+  static KeyTy pack(unsigned width, Domain domain) {
```

Comment:
What if `width = UINT_MAX`?
Should we at least assert this here to not silently drop a bit?

Reply:
Good point